resourcemanager/pooltask/task_manager_scheduler.go

```diff
@@ -16,10 +16,10 @@ package pooltask
 
 // Overclock is to increase the concurrency of pool.
 func (t *TaskManager[T, U, C, CT, TF]) Overclock() (tid uint64, task *TaskBox[T, U, C, CT, TF]) {
-	if t.concurrency > t.running.Load() {
-		return t.getBoostTask()
+	if t.running.Load()+1 >= int32(t.SubTaskCnt(tid)) {
+		return
 	}
-	return 0, nil
+	return t.getBoostTask()
```

Comment:
The old logic is not very clear. If tasks in the queue are ready to run, it is unnecessary to create a task.